### PR TITLE
Avoid instaiblity errors in code generated by `syntax_ext::deriving::call_intrinsic()`

### PR DESCRIPTION
**Pull request by jseyfried**
#35957 made old-style attribute syntax extensions no longer allow internal instability.

r? @eddyb

### CODE REVIEW

**Comment by jseyfried** [comment]
cc @nrc

File: src/libsyntax_ext/deriving/mod.rs [hunk 1]
```diff
@@ -229,7 +229,19 @@ fn expand_derive(cx: &mut ExtCtxt,
...
         } else {
             let name = intern_and_get_ident(&format!("derive_{}", tname));
-            let mitem = cx.meta_word(titem.span, name);
+            let span = Span {
+                expn_id: cx.codemap().record_expansion(codemap::ExpnInfo {
+                    call_site: titem.span,
+                    callee: codemap::NameAndSpan {
+                        format: codemap::MacroAttribute(intern(&format!("derive({})", tname))),
+                        span: Some(titem.span),
+                        allow_internal_unstable: true,
```
Comment (eddyb) [inline src/libsyntax_ext/deriving/mod.rs]: What about adding this block of code to `call_intrinsic` instead? It's about the same amount of work and wouldn't hide unstable feature usage like the previous implementation did.

Comment (jseyfried) [inline src/libsyntax_ext/deriving/mod.rs]: Good point, done.

**Comment by eddyb** [comment]
You could probably now replace `allow_internal_unstable: true` which is done for builtin derives.
But r=me either way, thanks!

**Comment by nrc** [comment]
cc @alexcrichton.

**Comment by alexcrichton** [comment]
Gah, sorry about the regression! It's not clear to me here what actually regressed, but the solution looks fine.

**Comment by jseyfried** [comment]
@alexcrichton no worries.
What  regressed was some "old-style" custom derives that were implemented using the framework in `libsyntax_ext/deriving/generic/mod.rs`, which sometimes uses `call_intrinsic`-generated code.

**Comment by alexcrichton** [comment]
Ah I see, that'd explain the lack of regressions in the main repo!

**Pull request merged**